src/core/thread/meshcop_dataset_manager.cpp

```diff
@@ -192,7 +193,31 @@ ThreadError DatasetManager::Register(void)
 
     VerifyOrExit((message = mSocket.NewMessage(0)) != NULL, error = kThreadError_NoBufs);
     SuccessOrExit(error = message->Append(header.GetBytes(), header.GetLength()));
-    SuccessOrExit(error = message->Append(mLocal.GetBytes(), mLocal.GetSize()));
+
+#if OPENTHREAD_ENABLE_COMMISSIONER
+    isCommissioner = mNetif.GetCommissioner().GetState() > MeshCoP::Commissioner::kStateDisabled ? true : false;
+#endif  // OPENTHREAD_ENABLE_COMMISSIONER
+
+    if (!isCommissioner)
+    {
+        SuccessOrExit(error = message->Append(mLocal.GetBytes(), mLocal.GetSize()));
+    }
+    else
+    {
+        const Tlv *cur = reinterpret_cast<const Tlv *>(mLocal.GetBytes());
+        const Tlv *end = reinterpret_cast<const Tlv *>(mLocal.GetBytes() + mLocal.GetSize());
+
+        while (cur < end)
+        {
+            Tlv::Type type = cur->GetType();
```

Comment:
A native commissioner sending a "Registration" message simply because the network is out-of-date should not include a Commissioner Session ID (i.e. it should act like a normal Thread device in this case).  However, if the Commissioner is sending an update like an external commissioner, then it should include a Commissioner Session ID.

In other words, a MGMT_ACTIVE_SET sent in response to a CLI command should include the Commissioner Session ID.  However, a MGMT_ACTIVE_SET sent in response to receiving an out-of-date Active Dataset from the network should not include the Commissioner Session ID.  Does that make sense?